### PR TITLE
feat(code-mappings): Use project platform to find Python projects

### PR DESCRIPTION
**Pull request by snigdhas**
Use the project platform instead of stacktrace files to identify Python projects

Followup from #40271

### CODE REVIEW

File: src/sentry/tasks/derive_code_mappings.py [hunk 1]
```diff
@@ -29,15 +29,19 @@ def identify_stacktrace_paths(
...
     if organizations is None:
         organizations = Organization.objects.filter(status=OrganizationStatus.ACTIVE)
 
     filename_maps = {}
     for org in organizations:
-        projects = Project.objects.filter(organization=org, first_event__isnull=False)
+        projects = Project.objects.filter(
+            organization=org,
+            first_event__isnull=False,
+            platform="python",
```
Comment (armenzg) [inline src/sentry/tasks/derive_code_mappings.py:43]: Is this the platform set by the customer on project creation? If that's the case we don't want this. I may have misunderstood your comment in Slack. I thought we were going to rely on the SDK's platform.

I assume a Group or the events may have a platform associated to it based from the SDK.

File: src/sentry/tasks/derive_code_mappings.py [hunk 2]
```diff
@@ -60,32 +64,24 @@ def get_all_stacktrace_paths(project: Project) -> List[str]:
     all_stacktrace_paths = set()
     for group in groups:
         event = group.get_latest_event()
-        is_python_stacktrace, stacktrace_paths = get_stacktrace_paths(event.data)
-        if not is_python_stacktrace:
-            return []
-        all_stacktrace_paths.update(stacktrace_paths)
+        all_stacktrace_paths.update(get_stacktrace_paths(event.data))
```
Comment (armenzg) [inline src/sentry/tasks/derive_code_mappings.py:67]: I would assume the platform may be extractable from the event.

File: src/sentry/tasks/derive_code_mappings.py (hunk 1 above)
Comment (snigdhas) [reply src/sentry/tasks/derive_code_mappings.py]: Ah good point, missed that distinction.

File: src/sentry/tasks/derive_code_mappings.py [hunk 3]
```diff
@@ -54,38 +57,32 @@ def identify_stacktrace_paths(
 
 def get_all_stacktrace_paths(project: Project) -> List[str]:
     groups = Group.objects.filter(
-        project=project, last_seen__gte=timezone.now() - GROUP_ANALYSIS_RANGE
+        project=project,
+        last_seen__gte=timezone.now() - GROUP_ANALYSIS_RANGE,
+        platform="python",
```
Comment (armenzg) [inline src/sentry/tasks/derive_code_mappings.py:62]: Ah OK. This is good.

**Review by armenzg — APPROVED** [review]
Fantastic!

**Pull request merged**